web/components/FeedFilterGroup.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+  Accordion,
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  Box,
+  Button,
+  Checkbox,
+  Flex,
+  Text,
+} from "@chakra-ui/react";
+import { Dispatch, SetStateAction } from "react";
+import Select from "react-select";
+import { PossibleTypes } from "../pages/onboarding";
+import { Filter, FilterGroup, SelectedFilters } from "./Feed";
+
+function FeedFilterGroup(props: {
+  key: string;
+  filterGroup: FilterGroup;
+  selectedFilters: SelectedFilters<Filter>;
+  setSelectedFilters: Dispatch<SetStateAction<SelectedFilters<Filter>>>;
+}) {
+  const { filterGroup, selectedFilters, setSelectedFilters } = props;
+
+  function updateFilters(filter: Filter, ind: number) {
+    let tempState = { ...selectedFilters };
+    if (tempState[filter.key].includes(filter.options[ind])) {
+      tempState[filter.key].splice(
+        tempState[filter.key].indexOf(filter.options[ind]),
+        1
+      );
+    } else {
+      tempState[filter.key].push(filter.options[ind]);
+    }
+    setSelectedFilters(tempState);
+    console.log(selectedFilters);
+  }
```

Comment:
Consider using [`useReducer`](https://react.dev/reference/react/useReducer) for your `selectedFilters` state. This will allow you to define a reducer function that outlines how to update to the new state based on the current one. Put this reducer function where in the file where your state resides.